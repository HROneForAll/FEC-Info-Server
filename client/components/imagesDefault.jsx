import React from 'react';
import styles from './styles.css.js';

const imagesDefault = (props) => {
  var count = 0;
  var toMap = {};
  for (var key in props.total) {
    if(props.total[key] && count <= props.toShow) {
      toMap[key] = props.total[key];
      count++;
    }
  }
  return (
    <div>
      <h1>Amenities</h1>
      {Object.keys(toMap).map((amenity) => (
        <div style={styles.initial}>
          <img style={styles.initial} name={amenity} src={props.images[amenity]}></img>
          <label htmlFor={amenity} style={styles.initial}>{amenity}</label>
        </div>
      ))}
    </div>
  );
};
export default imagesDefault;

