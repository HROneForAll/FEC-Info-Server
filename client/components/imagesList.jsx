import React from 'react';
import styles from './styles.css.js';

const imagesList = (props) => {

  return (
    <div style={props.show ? styles.modalShow : styles.modalHide}>
      <section style={styles.imageList}>
        <button onClick={() => {
          props.scroll();
          props.toggle();
        }}>Back</button>
        <h1 style={styles.present}>Special to this loft</h1>
        <p style={styles.present}>This home has these unique amenities</p>
        {Object.keys(props.special).map((amenity) => (
          props.special[amenity] 
            ? <div style={styles.special}>
              <label htmlFor={amenity} style={styles.present}>{amenity}
                <img name={amenity} src={props.images[amenity]} style={styles.present}></img>
              </label>
            </div> 
            : null
        ))}
        <h1 style={styles.present}>Essential Amenities</h1>
        <p style={styles.present}>This Airbnb Plus home comes with these amenities</p>
        {Object.keys(props.essential).map((amenity) => (
          props.essential[amenity] 
            ? <div style={styles.special}>
              <label htmlFor={amenity} style={styles.present}>{amenity}</label>
              <img name={amenity} src={props.images[amenity]} style={styles.present}></img>
            </div> 
            : null
        ))}
        <h1 style={styles.present}>Not available</h1>
        <p style={styles.present}>This home doesn't have these amenities</p>
        {Object.keys(props.special).map((amenity) => (
          props.special[amenity] 
            ? null 
            : <div style={styles.special}>
              <label htmlFor={amenity} style={styles.notPresent}>{amenity}
                <img name={amenity} src={props.images[amenity]} style={styles.notPresent}></img>
              </label>
            </div>
        ))}
        {Object.keys(props.essential).map((amenity) => (
          props.essential[amenity] 
            ? null 
            : <div style={styles.special}>
              <label className="essential" htmlFor={amenity} style={styles.notPresent}>{amenity}</label>
              <img name={amenity} src={props.images[amenity]} style={styles.notPresent}></img>
            </div>
        ))}
      </section>
    </div>
  );
};

export default imagesList;