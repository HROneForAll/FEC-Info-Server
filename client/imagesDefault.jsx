import React from 'react';
import styles from './styles.css.js';

const imagesDefault = (props) => {
  var count = 0; 
  return (
    <div>
      <h1>Amenities</h1>
      {Object.keys(props.special).map((amenity) => (
        props.special[amenity] 
          ? <div style={styles.initial}>
            <label className="special" htmlFor={amenity}>{amenity}</label>
            <img style={styles.initial} name={amenity} src={props.images[amenity]}></img>
          </div> 
          : null
      ))}
      {Object.keys(props.essential).map((amenity) => (
        props.essential[amenity] 
          ? <div style={styles.initial}>
            <label className="essential" htmlFor={amenity}>{amenity}</label>
            <img style={styles.initial} name={amenity} src={props.images[amenity]}></img>
          </div> 
          : null
      ))}
    </div>
  );
};
export default imagesDefault;