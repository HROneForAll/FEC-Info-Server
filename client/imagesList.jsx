import React from 'react';
import styles from './styles.css.js';

const imagesList = (props) => {

  return (
    <div style={props.show ? styles.modalShow : styles.modalHide}>
      <section style={styles.imageList}>
        <button onClick={() => props.toggle()}>Back</button>
        <h1>Special Amenities</h1>
        {Object.keys(props.special).map((amenity) => (
          props.special[amenity] 
            ? <div style={styles.special}>
              <label className="special" htmlFor={amenity} style={styles.present}>{amenity}
                <img name={amenity} src={props.images[amenity]} style={styles.present}></img>
              </label>
            </div> 
            : null
        ))}
        <h1>Essential Amenities</h1>
        {Object.keys(props.essential).map((amenity) => (
          props.essential[amenity] 
            ? <div style={styles.special}>
              <label className="essential" htmlFor={amenity} style={styles.present}>{amenity}</label>
              <img name={amenity} src={props.images[amenity]} style={styles.present}></img>
            </div> 
            : null
        ))}
        <h1>Unavailable Amenities</h1>
        {Object.keys(props.special).map((amenity) => (
          props.special[amenity] 
            ? null 
            : <div style={styles.special}>
              <label className="special" htmlFor={amenity} style={styles.notPresent}>{amenity}
                <img name={amenity} src={props.images[amenity]} style={styles.notPresent}></img>
              </label>
            </div>
        ))}
        {Object.keys(props.essential).map((amenity) => (
          props.essential[amenity] 
            ? null 
            : <div style={styles.special}>
              <label className="essential" htmlFor={amenity} style={styles.notPresent}>{amenity}</label>
              <img name={amenity} src={props.images[amenity]} style={styles.notPresent}></img>
            </div>
        ))}
      </section>
    </div>
  );
};

export default imagesList;