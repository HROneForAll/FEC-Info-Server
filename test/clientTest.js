import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Amenities from '../client/components/amenities.jsx';
import imagesDefault from '../client/components/imagesDefault.jsx';
import imagesList from '../client/components/imagesList.jsx';

describe('Amenities', () => {
  it('renders on the page', () => {
    const wrapper = shallow(<Amenities />);
    const component = wrapper.shallow();

    expect(toJson(component)).toMatchSnapshot();
  });

  it()
})
