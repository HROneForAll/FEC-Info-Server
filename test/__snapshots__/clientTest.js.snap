// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amenities renders on the page 1`] = `
<div>
  <imagesList
    essential={Object {}}
    images={Object {}}
    show={false}
    special={Object {}}
    toggle={[Function]}
  />
  <imagesDefault
    essential={Object {}}
    images={Object {}}
    special={Object {}}
  />
  <button
    onClick={[Function]}
  >
    Show All Amenities
  </button>
</div>
`;

exports[`Default Image List renders on the page 1`] = `<imagesDefault />`;

exports[`Modal Image List renders on the page 1`] = `<imagesList />`;
